src/Service/EntityMetadata.php

```diff
@@ -178,15 +203,94 @@ public function extractExposedProperties(\ReflectionClass $reflection)
         return $this->exposedPropertiesForClass[$className];
     }
 
+    /**
+     * Get the ID property for a class
+     */
+    public function extractId(ReflectionClass $reflection): string
+    {
+        $className = $reflection->getName();
+        if (!array_key_exists($className, $this->idForClasses)) {
+            $properties = $reflection->getProperties();
+
+            $ids = array_filter($properties, function (ReflectionProperty $property) {
+                $annotation = $this->annotationReader->getPropertyAnnotation(
+                    $property,
+                    Id::class
+                );
+
+                return !is_null($annotation);
+            });
+            if (!$ids) {
+                throw new \Exception("${className} has no property annotated with @Id");
```

Comment:
might as well import the FQN for `Exception`.

Reply:
I'll make a note and fix this in my next pass (which is incoming as soon as this is merged).